doop-image-checker/main.go

```diff
@@ -101,15 +103,46 @@ func handleHeaders(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	//make request to Keppel's Registry API (we deliberately make a GET here in
-	//order to explicitly update the image's last_pulled_at timestamp);
-	//unfortunately we need to engage in some trickery to extract the
-	//X-Keppel-Vulnerability-Status header
 	var hc headerCapturer
```

Comment:
I don't like that `headerCapturer` gets used for the new code path with the hardcoded values, even though it is a special structure that very specifically applies to the case of talking to the actual Registry API only.  The type `headerCapturer` should be confined to the latter branch, whereas the new code path should rather use only `http.Header`.